lib/view/postcard_chat.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:convert';
+
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/model/pair.dart';
+import 'package:autonomy_flutter/screen/chat/chat_thread_page.dart';
+import 'package:autonomy_flutter/service/configuration_service.dart';
+import 'package:autonomy_flutter/service/chat_service.dart';
+import 'package:autonomy_flutter/util/asset_token_ext.dart';
+import 'package:autonomy_flutter/util/constants.dart';
+import 'package:autonomy_flutter/util/datetime_ext.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:autonomy_flutter/util/style.dart';
+import 'package:autonomy_flutter/view/tappable_forward_row.dart';
+import 'package:autonomy_theme/autonomy_theme.dart';
+import 'package:autonomy_theme/extensions/theme_extension/moma_sans.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:libauk_dart/libauk_dart.dart';
+import 'package:nft_collection/models/asset_token.dart';
+import 'package:uuid/uuid.dart';
+import 'package:autonomy_flutter/model/chat_message.dart' as app;
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+
+class MessagePreview extends StatefulWidget {
+  final MessagePreviewPayload payload;
+
+  const MessagePreview({Key? key, required this.payload}) : super(key: key);
+
+  @override
+  State<MessagePreview> createState() => _MessagePreviewState();
+}
+
+class _MessagePreviewState extends State<MessagePreview> {
+  final ChatService _postcardChatService = injector<ChatService>();
+  Pair<WalletStorage, int>? _wallet;
+  app.Message? _lastMessage;
+  late AssetToken _assetToken;
+  int _newMessageCount = 0;
+  final String _fetchId = const Uuid().v4();
+  late final ChatListener _chatListener;
+  bool _didFetch = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _assetToken = widget.payload.asset;
+    _websocketInit();
+  }
+
+  Future<void> _websocketInit() async {
+    _newMessageCount = 0;
+    final wallet = await widget.payload.asset.getOwnerWallet();
+    if (wallet == null) {
+      return;
+    }
+    _wallet = wallet;
+    final address = widget.payload.asset.owner;
+    final id = widget.payload.asset.id;
+    await _postcardChatService.connect(
+        address: address, id: id, wallet: wallet);
+    _chatListener = ChatListener(
```

Comment:
I think `ChatListener` can be assigned right when defining the variable. No need to use late, it can lead to an unexpected crash if the `await` above takes too long then the user goes back before it's done and the variable was used before initiating.

Reply:
ChatListener( uses instance member so it can not be accessed during initialize.
To cover you case, i propose to use nullable object instead of late